### PR TITLE
feat: Added erfinv in paddle frontend with test

### LINKED ISSUE #23710: erfinv

### PR DESCRIPTION
**Pull request by RummanAli**
# Related Issue
closes #23710 

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [ ] Did you run your tests and are your tests passing?
- [ ] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by RummanAli** [comment]
@hello-fri-end I need some help regarding the tests that are failing.

**Comment by RummanAli** [comment]
[erfinv functional api](https://github.com/unifyai/ivy/issues/23685) Someone has already opened the backend issue!

**Comment by RummanAli** [comment]
closes #23710

File: ivy/functional/frontends/paddle/math.py [hunk 1]
```diff
@@ -193,6 +193,11 @@ def divide(x, y, name=None):
 def erf(x, name=None):
     return ivy.erf(x)
 
+@with_supported_dtypes({"2.5.1 and below": ("float32", "float64")}, "paddle")
+@to_ivy_arrays_and_back
+def erfinv(x, name=None):
+    return  ivy.erfinv
```
Comment (hello-fri-end) [inline ivy/functional/frontends/paddle/math.py:199]: you are returning a function reference here. Do you mean to do `ivy.erfinv(x)` here ( provided that `ivy.erfinv` exists)?

**Review by hello-fri-end — CHANGES_REQUESTED** [review]
Hey @RummanAli ! Thanks for creating this PR. Could you please resolve the comment I have left? I think that should fix your failing tests. Happy to clarify anything furter: )_

File: ivy/functional/frontends/paddle/math.py (hunk 1 above)
Comment (RummanAli) [reply ivy/functional/frontends/paddle/math.py:199]: Yea my bad let me fix it, also I think ivy.erfinv does not exist as of now : [erfinv-functional](https://github.com/unifyai/ivy/issues/23685). So is there a way to bypass it as someone is already working to implement erfinv in functional.

File: ivy/functional/frontends/paddle/math.py [hunk 2]
```diff
@@ -193,6 +193,11 @@ def divide(x, y, name=None):
 def erf(x, name=None):
     return ivy.erf(x)
 
+@with_supported_dtypes({"2.5.1 and below": ("float32", "float64")}, "paddle")
+@to_ivy_arrays_and_back
+def erfinv(x, name=None):
+    return  ivy.erfinv(x)
```
Comment (hello-fri-end) [inline ivy/functional/frontends/paddle/math.py:199]: There's no `ivy.erfinv` function. You will need to implement `erfinv` using a composition of other `ivy` functions.

**Review by hello-fri-end — CHANGES_REQUESTED** [review]
Hey @RummanAli, it seems you are trying to use a function in your implementation which does not exist. You should implement `erfinv` as a composition of other ivy functions. Happy to clarify anything furher. Thanks :)

**Comment by Aaryan562** [comment]
@RummanAli @hello-fri-end should I take this issue?

**Comment by RummanAli** [comment]
> @RummanAli @hello-fri-end should I take this issue?

I am currently working on it!😄

**Comment by RummanAli** [comment]
@hello-fri-end Thanks for your review, I have implemented erfinv using composition of ivy functions and 80% of tests are passing:
image
Can you give me leads as to how I can resolve this cpu error.

**Comment by RummanAli** [comment]
Also open for any suggestions that might increase its efficiency, Tried fancy indexing but it fails for some of the backends.

**Review by hello-fri-end — CHANGES_REQUESTED** [review]
Hey @RummanAli ! Thanks for taking the time to work on this change. Actually, we have a policy for keeping the [frontend implementations short](https://unify.ai/docs/ivy/overview/deep_dive/ivy_frontends.html#short-frontend-implementations). I didn't expect that the compositional implementation will be more than a few lines, my apologies for not realizing this earlier.  We should be adding the `erfinv` function to our experimental API so that the frontend implementation is only a few lines. You should first create an issue in our repo using this [template](https://unify.ai/docs/ivy/overview/contributing/open_tasks.html#creating-an-issue-on-ivy-s-github-using-a-template) guide. A memeber of our team will team review the request and add it to an open task. Then it's your choice, if you want to take it up yourself and wait for someelse else to pick it up. Happy to clarify anything further in this regard :)

**Comment by RummanAli** [comment]
> Hey @RummanAli ! Thanks for taking the time to work on this change. Actually, we have a policy for keeping the [frontend implementations short](https://unify.ai/docs/ivy/overview/deep_dive/ivy_frontends.html#short-frontend-implementations). I didn't expect that the compositional implementation will be more than a few lines, my apologies for not realizing this earlier. We should be adding the `erfinv` function to our experimental API so that the frontend implementation is only a few lines. You should first create an issue in our repo using this [template](https://unify.ai/docs/ivy/overview/contributing/open_tasks.html#creating-an-issue-on-ivy-s-github-using-a-template) guide. A memeber of our team will team review the request and add it to an open task. Then it's your choice, if you want to take it up yourself and wait for someelse else to pick it up. Happy to clarify anything further in this regard :)

Okkk thanks for reviewing!

**Comment by RummanAli** [comment]
> Hey @RummanAli ! Thanks for taking the time to work on this change. Actually, we have a policy for keeping the [frontend implementations short](https://unify.ai/docs/ivy/overview/deep_dive/ivy_frontends.html#short-frontend-implementations). I didn't expect that the compositional implementation will be more than a few lines, my apologies for not realizing this earlier. We should be adding the `erfinv` function to our experimental API so that the frontend implementation is only a few lines. You should first create an issue in our repo using this [template](https://unify.ai/docs/ivy/overview/contributing/open_tasks.html#creating-an-issue-on-ivy-s-github-using-a-template) guide. A memeber of our team will team review the request and add it to an open task. Then it's your choice, if you want to take it up yourself and wait for someelse else to pick it up. Happy to clarify anything further in this regard :)

Somebody has already opened this issue:
https://github.com/unifyai/ivy/issues/23685

**Comment by hello-fri-end** [comment]
> > Hey @RummanAli ! Thanks for taking the time to work on this change. Actually, we have a policy for keeping the [frontend implementations short](https://unify.ai/docs/ivy/overview/deep_dive/ivy_frontends.html#short-frontend-implementations). I didn't expect that the compositional implementation will be more than a few lines, my apologies for not realizing this earlier. We should be adding the `erfinv` function to our experimental API so that the frontend implementation is only a few lines. You should first create an issue in our repo using this [template](https://unify.ai/docs/ivy/overview/contributing/open_tasks.html#creating-an-issue-on-ivy-s-github-using-a-template) guide. A memeber of our team will team review the request and add it to an open task. Then it's your choice, if you want to take it up yourself and wait for someelse else to pick it up. Happy to clarify anything further in this regard :)
> 
> Somebody has already opened this issue: #23685

In this case, let's wait for their PR to get merged first. Feel free to pick up other tasks in the meantime :)

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**